include/dxc/DXIL/DxilInstructions.h

```diff
@@ -645,6 +645,42 @@ struct LlvmInst_VAArg {
   bool isAllowed() const { return false; }
 };
 
+/// This instruction extracts from vector
+struct LlvmInst_ExtractElement {
+  llvm::Instruction *Instr;
+  // Construction and identification
+  LlvmInst_ExtractElement(llvm::Instruction *pInstr) : Instr(pInstr) {}
+  operator bool() const {
+    return Instr->getOpcode() == llvm::Instruction::ExtractElement;
+  }
+  // Validation support
+  bool isAllowed() const { return true; }
```

Comment:
A little suprised to see `ExtractElement`\`InsertElement`\`ShuffleVector` weren't already  defined since I expected at least the first two as required for scalarization which DXIL already supports.

If these are new for vectorization, should` isAllowed()` be checking `IsSM69Plus()` instead of always returning true?